noxfile.py

```diff
@@ -100,53 +123,62 @@ def test_sphinx(session: nox.Session, sphinx: int) -> None:
-    """Translation commands. Available commands after `--` : extract, update, compile, init."""
-    # get the command from posargs, default to "update"
-    pybabel_cmd, found = ("update", False)
-    for c in ["extract", "update", "compile", "init"]:
-        if c in session.posargs:
-            pybabel_cmd, found = (c, True)
-
-    if found is False:
-        print(
-            "No translate command found. Use like: `nox -s translate -- COMMAND`."
-            "\ndefaulting to `update`"
-            "\nAvailable commands: extract, update, compile, init"
-        )
-
-    # get the language from parameters default to en.
-    # it can be deceiving but we don't have a table of accepted languages yet
-    lan = "en" if len(session.posargs) < 2 else session.posargs[-1]
-
-    # get the path to the differnet local related pieces
-    locale_dir = str(ROOT / "src" / "pydata_sphinx_theme" / "locale")
-    babel_cfg = str(ROOT / "babel.cfg")
-    pot_file = str(locale_dir / "sphinx.pot")
-
-    # install deps
+    """Update translation related files."""
+    session.install("Babel", "jinja2")
+
+    # fmt: off
+    session.run(  # generate/update the .pot file
+        "pybabel", "extract", ".",
+        "-F", str(babel_cfg.relative_to(root_dir)),
+        "-o", str(pot_file.relative_to(root_dir)),
+        #"-k", "'_ __ l_ lazy_gettext'",
+    )
+    # fmt: on
+
+    # update the message catalog (.po)
+    languages = [f.stem for f in locale_dir.glob("*") if f.is_dir()]
+    # fmt: off
+    cmd = [
+        "pybabel", "update",
+        "-i", str(pot_file.relative_to(root_dir)),
+        "-d", str(locale_dir.relative_to(root_dir)),
+        "-D", "sphinx"
+    ]
+    # fmt: on
+    for lan in languages:
+        session.run(*cmd, "-l", lan)
+
+
+@session(default=False)
+def add_language(session: nox.Session) -> None:
+    """Add a language to the catalog using posargs."""
     session.install("Babel")
 
-    # build the command from the parameters
-    cmd = ["pybabel", pybabel_cmd]
-
-    if pybabel_cmd == "extract":
-        cmd += [ROOT, "-F", babel_cfg, "-o", pot_file, "-k", "_ __ l_ lazy_gettext"]
-
-    elif pybabel_cmd == "update":
-        cmd += ["-i", pot_file, "-d", locale_dir, "-D", "sphinx"]
-
-    elif pybabel_cmd == "compile":
-        cmd += ["-d", locale_dir, "-D", "sphinx"]
-
-    elif pybabel_cmd == "init":
-        cmd += ["-i", pot_file, "-d", locale_dir, "-D", "sphinx", "-l", lan]
-
-    session.run(cmd)
-
-
-@nox.session()
+    # get the language name from posargs
+    if len(session.posargs) != 1:
+        raise ValueError(
+            f"There should be only one posargs: the ISO code of the languge, {len(session.posargs)} given."
+        )
+    lan = session.posargs[0]
+
+    # init new language
+    session.run(
+        "pybabel",
+        "init",
+        "-i",
+        str(pot_file),
+        "-d",
+        str(locale_dir),
+        "-D",
+        "sphinx",
+        "-l",
+        lan,
```

Comment:
can we wrap this in `# fmt: off` too?  I think it's helpful when reading the code to keep the flag and its associated value on the same line.